### PR TITLE
FIX #3: Update Annoy version

### LINKED ISSUE #3: Update annoy version
Hi,

Thanks for writing this library!

I want to use your bindings but AFAIU it contains two years old version of annoy and I need some recent feature (Manhatan . Are you interested in upgrading (we want to provide PR's)?

I'm not an node.js expert, but I wonder if we could move annoy code outside of this bindings (like downloading it during installation), so it will be clear which version we are building against and the update path would be somewhat easier (making diffs between annoy versions etc.). Do you think this is good idea?

### PR DESCRIPTION
**Pull request by mbuszka**
- Update Annoy library to current master
- Add Manhattan metric option to wrapper
- getDistance now returns proper distance, not squared
- fixed tests, added test for manhattan distance

### CODE REVIEW

File: kissrandom.h [hunk 1]
```diff
@@ -0,0 +1,106 @@
+#ifndef KISSRANDOM_H
+#define KISSRANDOM_H
+
```
Comment (jimkang) [inline kissrandom.h:3]: Interesting! Are there any obviously noticeable effects of using this instead of RandRandom?

**Review by jimkang — APPROVED** [review]
Thanks so much for your hard work! I'll try it out and merge if it seems fine.

**Comment by jimkang** [comment]
Published in NPM as version 2.0.

File: kissrandom.h (hunk 1 above)
Comment (mbuszka) [reply kissrandom.h:3]: I didn't notice, but it's used upstream, so i thought we should use it as well to keep consistency. (Also they removed RandRandom)

**Pull request merged**